sdk/iot/core/src/az_iot_sas_token.c

```diff
@@ -25,9 +25,9 @@ az_result az_iot_sas_token_get_document(
     az_span document,
     az_span* out_document)
 {
-  AZ_PRECONDITION_VALID_SPAN(device_id, 1);
-  AZ_PRECONDITION_VALID_SPAN(iothub_fqdn, 1);
-  AZ_PRECONDITION_VALID_SPAN(document, 0);
+  AZ_PRECONDITION_VALID_SPAN(device_id, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(iothub_fqdn, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(document, 0, false);
```

Comment:
Are you sure document can't be NULL span?

Reply:
I don't believe so. The API attempts to append data to the span so writing to a NULL buffer shouldn't be allowed.